peekaboo/sample.py

```diff
@@ -405,38 +402,10 @@ def file_extension(self):
         return self.__file_extension
 
     @property
-    def mimetypes(self):
-        """ Determines the mimetypes of this sample. """
-        if self.__mimetypes:
-            return self.__mimetypes
-
-        mime_types = set()
-
+    def declared_mimetype(self):
+        """ Determines the mimetype of this sample. """
         # get MIME type from meta info
-        declared_mt = None
-        if self.meta_info_type_declared:
-            declared_mt = self.meta_info_type_declared
-            if declared_mt is not None:
-                logger.debug('Sample declared as "%s"' % declared_mt)
-                mime_types.add(declared_mt)
-
-        declared_filename = self.__filename
```

Comment:
Uh, this woke me up: Here we base the extension-based type determination on the file name used for submission but prefer a declared name supplied by the client if available. Where does this go in the new concept? Do we want so save this anyway?

From what I can tell, FileTools always uses `Sample.filename` which is always the client's submission file name but not the declared name. Should we have two properties `FileToolsReport.type_by_name` (i.e. by `Sample.filename`) and `FileToolsReport.type_by_declared_name` (i.e. `Sample.name_declared`).

This comment in `Sample.file_extension` might also be relevant here:
```
        # try to find a file name containing an extension. Using
        # self.__filename will almost never yield anything useful because
        # amavis intentionally hands us files named only p001, p002 and so on.
        # But we still try it in case there's no declared name.
        filename = self.__filename
        if self.name_declared:
            filename = self.name_declared
```

Based on this, the current implementation would, at least with AMaViSes `p001` files never yield anything useful.
Should we also have a test for that?

Reply:
This is now also handled in #143 by reinstating the lookup and use of `name_declared` in the sample: https://github.com/scVENUS/PeekabooAV/pull/143/commits/188465f3061d88a9680de678268ac077565d7d65.